File: OrientationPFIEV/Models/Choix.cs
namespace OrientationPFIEV.Models;

/// <summary>
/// A candidate's program preference choice.
/// NumeroDeChoix: 1 = first choice, 2 = second, etc.
/// ChoixAdmis: true when the candidate is admitted through this choice.
/// </summary>
public class Choix
{
    public int ChoixID { get; set; }
    public int CandidatID { get; set; }
    public int FiliereID { get; set; }
    public int NumeroDeChoix { get; set; }
    public bool ChoixAdmis { get; set; }
}

File: app/models/choix.py
from dataclasses import dataclass


@dataclass
class Choix:
    choix_id: int = 0
    candidat_id: int = 0
    filiere_id: int = 0
    choix_ordre: int = 1
    choix_admis: bool = False
    filiere_nb_place: int = 0
